specs/deque.spec.js

```diff
@@ -0,0 +1,154 @@
+const Deque = require('../src/deque');
+
+describe('Deque', () => {
+    describe('Deque Constructor', () => {
+        it('Create new Deque', () => {
+            const Deque = new Deque();
+
+            assert.equal(Deque.data, []);
+        });
+    });
+
+    describe('Deque push function', () => {
+        beforeEach( () => { 
+            const Deque = new Deque();
+            const arr = [1, 2, 3, 4, 5];
+            Deque.push(true, arr);
+        });
+
+        it('Push items to the clean Deque', () => {
+            assert.equal(Deque.data, [1, 2, 3, 4, 5]);
+        });
+
+        it('Push items with left = false argument', () => {
+            const arr2 = [10, 20, 30];
+
+            Deque.push(false, arr2);
+
+            assert.equal(Deque.data, [10, 20, 30, 1, 2, 3, 4, 5]);
+        });
+
+        it('Push items with left = true argument', () => {
+            const arr2 = [10, 20, 30];
+
+            Deque.push(true, arr2);
+
+            assert.equal(Deque.data, [1, 2, 3, 4, 5, 10, 20, 30]);
+        });
+
+    });
+
+    describe('Deque pop function', () => {
+        beforeEach( () => { 
+            const Deque = new Deque();
+            const arr = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+            const left = true;
+            const right = false;
+
+            Deque.push(left, arr);
+        });
+        
+        it('Pop n first(left = true) items', () => {
+            const n = 5;
+
+            const returned = Deque.pop(n, left);
+
+            assert.equal(returned, [0, 1, 2, 3, 4]);
+            assert.equal(Deque.data, [5, 6, 7, 8, 9]);
```

Comment:
mixing scenarios, better to make 2 test: data is poped, some data remain in deque.